contracts/lease/src/contract/cmd/open.rs

```diff
@@ -61,18 +64,19 @@ impl<'a> WithLppLender for OpenLoanReq<'a> {
             },
         )
         .ok_or_else(Self::Error::NoPaymentError)??;
+
         if downpayment_lpn.is_zero() {
-            Err(Self::Error::InsufficientPayment(downpayment))
-        } else {
-            let borrow_lpn = self.liability.init_borrow_amount(downpayment_lpn);
+            return Err(Self::Error::InsufficientPayment(downpayment));
+        }
 
-            lpp.open_loan_req(borrow_lpn)?;
+        let borrow_lpn = self.liability.init_borrow_amount(downpayment_lpn);
 
-            Ok(Self::Output {
-                batch: lpp.into().batch,
-                downpayment,
-            })
-        }
+        lpp.open_loan_req(self.max_loan.map_or(borrow_lpn, Coin::new).min(borrow_lpn))?;
```

Comment:
perform the calculation as part of the `.init_borrow_amount(downpayment_lpn, max_ltv)`. 

Hint:
Apply `self.initial.max(max_ltv)` as a first step in the `fn Liability::init_borrow_amount`.

Update the current and add sufficient unit tests there.